sql/2025/performance/web_vitals_by_technology.sql

```diff
@@ -0,0 +1,275 @@
+CREATE TEMP FUNCTION IS_GOOD(good FLOAT64, needs_improvement FLOAT64, poor FLOAT64) RETURNS BOOL AS (
+  SAFE_DIVIDE(good, (good + needs_improvement + poor)) >= 0.75
+);
+
+CREATE TEMP FUNCTION IS_POOR(good FLOAT64, needs_improvement FLOAT64, poor FLOAT64) RETURNS BOOL AS (
+  SAFE_DIVIDE(poor, (good + needs_improvement + poor)) >= 0.25
+);
+
+CREATE TEMP FUNCTION IS_NI(good FLOAT64, needs_improvement FLOAT64, poor FLOAT64) RETURNS BOOL AS (
+  NOT IS_GOOD(good, needs_improvement, poor) AND
+  NOT IS_POOR(good, needs_improvement, poor)
+);
+
+CREATE TEMP FUNCTION IS_NON_ZERO(good FLOAT64, needs_improvement FLOAT64, poor FLOAT64) RETURNS BOOL AS (
+  good + needs_improvement + poor > 0
+);
+
+WITH base AS (
+  SELECT
+    date,
+    origin,
+    CONCAT(origin, '/') AS page,
+    CASE
+      WHEN device = 'phone' THEN 'mobile'
+      ELSE device
+    END AS client,
+
+    # TODO: Should we remove FID now that it's deprecated?
+    fast_fid,
+    avg_fid,
+    slow_fid,
```

Comment:
❓ Now that FID is deprecated, should I go ahead and remove FID references from these queries?

Reply:
Yes, since we no longer populate these in CrUX. So they just return `null`.